csv_importer/csv_importer/db_manage.py

```diff
@@ -0,0 +1,199 @@
+    are abbreviations, the values are SubjectItem instances. Such dictionary
+    is used in refince_csv for mapping the found subject abbreviation in certain
+    CSV to the choosen one.
+
+    In the database there is a table Subject which has the same columnd - id,
+    name and abbreviations. This table is used as source of information
+    for SubjectItem instances.
+    Check the function load_subject_abbr_map() below.
+
+    The foreign languages subjects (English, Spanish, Russion, etc.) have
+    variations with different levels - Б1, Б1.1 and Б2. They are treated as
+    different subject items, so in the Subject table there are records for
+    'АЕ', 'АЕ-Б1', 'АЕ-Б1.1', 'АЕ-Б2', and similar records for all other foreign
+    languagues.
+
+    Below in the init_db() function there's is manually initialized list
+    of SubjectItems which is used to fill the Subject table in the database.
+
+    """
+    id: str
+    name: str
+    abbreviations: list[str]
+
+    def __post_init__(self):
+        self.id = self.id.upper()
+
+    def raw_strings(self):
+        result = []
+        for a in [self.id, *self.abbreviations]:
+            result.append(f'{a.lower()}')
+        return result
+
+
+def load_subject_abbr_map() -> dict[str, SubjectItem]:
+    """
+    Returns a mapping where the keys are subject abbreviations and the
+    values are SubjectItem instances. Since one subject could have multiple
+    abbreviations, the same SubjectItem could participate in multiple paris
+    in the result dictionary.
+
+    The abbreviations (the keys) are lowercased.
+
+    Check the SubjectItem docs for more details.
+    """
+    db = get_db_engine()
+    with Session(db) as session:
+        records = session.execute(select(Subject))
+        subject_items = []
+        for r in records:
+            abbreviations = [] if not r[2] else r[2].split(',')
+            subject_items.append(SubjectItem(
+                id=r[0], name=r[1], abbreviations=abbreviations
+            ))
+
+        result = {}
+        for item in subject_items:
+            for raw_str in item.raw_strings():
+                result[raw_str] = item
+
+        return result
+
+
+def init_db():
+    """
+    Initializes a new Database.
+    Currently it fills the Subject table.
+    TODO: This could be implemented as alembic migration.
+    """
+
+    language_levels = ['Б1', 'Б1.1', 'Б2']
+
+    def _variations(this: SubjectItem, variations: list[str]) -> list[SubjectItem]:
+        result = [this]
+        for v in variations:
+            result.append(
+                SubjectItem(
+                    id=f'{this.id}-{v}',
+                    name=f'{this.name} {v}',
+                abbreviations=[f'{a}-{v}' for a in this.abbreviations]
+                )
+            )
+
+        return result
+
+
+    def _lang_variations(this: SubjectItem) -> list[SubjectItem]:
+        return _variations(this, language_levels)
+
+    default_subject_items = [
```

Comment:
Just to check my understanding, if a new abbreviation appears in the raw data in the future, should we come in an manually add it to the respective `abbreviations` list in `default_subject_items`?

Reply:
That's correct. 
Also in order the new item to appear in the DB one should execute the following command:
`DB_URL="postgresql://....." ./csv_importer/app.py init-db`
Probably this part can be improved